### PR TITLE
Like command

### PR DESCRIPTION
**Pull request by StavZ**
I added a like command to pc-spotify plugin which allows add current playing song to liked songs

### CODE REVIEW

**Review by cyyynthia — CHANGES_REQUESTED** [review]
The command doesn't take into account if the account is from Discord or Powercord. We can only update someone's library if they linked their account to Powercord since we need some extra scopes Discord doesn't request. See [this](https://github.com/powercord-org/powercord/blob/v2-dev/src/Powercord/plugins/pc-spotify/components/Modal.jsx#L160-L168) for an example on how to check this.

**Comment by StavZ** [comment]
Ok. I'll fix it

File: src/Powercord/plugins/pc-spotify/commands/like.js [hunk 1]
```diff
@@ -12,6 +12,14 @@ module.exports = {
         result: 'You are not currently listening to anything.'
       };
     }
+    const hasCoolFeatures = powercord.account && powercord.account.spotify;
+    if (!hasCoolFeatures) {
+      return {
+        send: false,
+        result: 'You need a [powercord](https://powercord.dev/) account and connected Spotify account.'
```
Comment (davidcralph) [inline src/Powercord/plugins/pc-spotify/commands/like.js]: Powercord should be capitalised here 

(i just realised i commented on an outdated version of the file, but this issue still applies)

**Review by davidcralph — APPROVED** [review]

File: src/Powercord/plugins/pc-spotify/commands/like.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        result: 'You are not currently listening to anything.'
+      };
+    }
+    const hasCoolFeatures = powercord.account && powercord.account.spotify;
+    if (!hasCoolFeatures) {
+      return {
+        send: false,
+        result: 'You need a Powercord account and connected Spotify account.'
+      };
+    }
```
Comment (cyyynthia) [inline src/Powercord/plugins/pc-spotify/commands/like.js]: It'd be better to check for the Powercord account first
```suggestion
    if (!powercord.account && powercord.account.spotify) {
      return {
        send: false,
        result: 'You need a Powercord account and connected Spotify account.'
      };
    }
    const currentTrack = playerStore.getCurrentTrack();
    if (!currentTrack) {
      return {
        send: false,
        result: 'You are not currently listening to anything.'
      };
    }
```

File: src/Powercord/plugins/pc-spotify/commands/like.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    const { body } = await SpotifyAPI.checkLibrary(currentTrack.id);
+    if (body[0]) {
+      SpotifyAPI.removeSong(currentTrack.id);
+      return {
+        send: false,
+        result: `You removed **${currentTrack.name} by ${currentTrack.artists}** from your Liked Songs`
```
Comment (cyyynthia) [inline src/Powercord/plugins/pc-spotify/commands/like.js]: Styling nitpicking
```suggestion
        result: `You removed **${currentTrack.name}** by **${currentTrack.artists}** from your Liked Songs`
```

File: src/Powercord/plugins/pc-spotify/commands/like.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      };
+    }
+    SpotifyAPI.addSong(currentTrack.id);
+    return {
+      send: false,
+      result: `You added **${currentTrack.name} by ${currentTrack.artists}** to your Liked Songs`
```
Comment (cyyynthia) [inline src/Powercord/plugins/pc-spotify/commands/like.js]: Styling nitpicking
```suggestion
      result: `You added **${currentTrack.name}** by **${currentTrack.artists}** to your Liked Songs`
```

**Review by cyyynthia — CHANGES_REQUESTED** [review]

**Review by cyyynthia — APPROVED** [review]

**Pull request merged**